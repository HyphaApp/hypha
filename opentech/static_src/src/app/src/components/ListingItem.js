import React from 'react';
import PropTypes from 'prop-types';


export default class ListingItem extends React.Component {
    render() {
        return (
            <li className="listing__item">
                <a className="listing__link">{this.props.item.title}</a>
            </li>
        );
    }
}

ListingItem.propTypes = {
    item: PropTypes.shape({
        title: PropTypes.string,
    }),
};
