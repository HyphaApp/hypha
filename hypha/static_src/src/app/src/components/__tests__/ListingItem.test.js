import React from "react";
import { mount } from "enzyme";
import sinon from "sinon";
import ListingItem from "../ListingItem";
import * as enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
enzyme.configure({ adapter: new Adapter() });

describe("Test Listing item component with selected", () => {
    const item = {title : "Title Text"}
    const selected = true
    const onClick = jest.fn()

    const subject = mount(<ListingItem item={item} 
        selected={selected} onClick={onClick}/>
    );
  
    it("Shoud render without issues", () => {
      expect(subject.length).toBe(1);
    });
  
    it("Should have searched classes and elements", () => {
      expect(subject.find('.is-active').length).toBe(1);
      subject.find('.listing__link').props().onClick();
      expect(onClick).toHaveBeenCalled()
      expect(subject.find('a').text()).toBe(item.title);
    });
  
    test('render a listing item component', () => {
      expect(subject).toMatchSnapshot();
    });
  
  });

  describe("Test Listing item component", () => {
    const item = {title : "Title Text"}
    const selected = false
    const onClick = jest.fn()

    const subject = mount(<ListingItem item={item} 
        selected={selected} onClick={onClick}/>
    );
  
    it("Shoud render without issues", () => {
      expect(subject.length).toBe(1);
    });
  
    it("Should have searched classes and elements", () => {
      expect(subject.find('.is-active').length).toBe(0);
      subject.find('.listing__link').props().onClick();
      expect(onClick).toHaveBeenCalled()
      expect(subject.find('a').text()).toBe(item.title);
    });
  
    test('render a listing item component', () => {
      expect(subject).toMatchSnapshot();
    });
  
  });
