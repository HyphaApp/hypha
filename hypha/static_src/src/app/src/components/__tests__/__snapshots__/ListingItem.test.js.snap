// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Listing item component render a listing item component 1`] = `
<ListingItem
  item={
    Object {
      "title": "Title Text",
    }
  }
  onClick={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
    }
  }
  selected={false}
>
  <li
    className="listing__item "
  >
    <a
      className="listing__link"
      onClick={
        [MockFunction] {
          "calls": Array [
            Array [],
          ],
        }
      }
    >
      Title Text
    </a>
  </li>
</ListingItem>
`;

exports[`Test Listing item component with selected render a listing item component 1`] = `
<ListingItem
  item={
    Object {
      "title": "Title Text",
    }
  }
  onClick={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
    }
  }
  selected={true}
>
  <li
    className="listing__item is-active"
  >
    <a
      className="listing__link"
      onClick={
        [MockFunction] {
          "calls": Array [
            Array [],
          ],
        }
      }
    >
      Title Text
    </a>
  </li>
</ListingItem>
`;
